src/core/src/preprocess/preprocess_steps_impl.cpp

```diff
@@ -428,6 +428,63 @@ void PreStepsList::add_convert_color_impl(const ColorFormat& dst_format) {
                 context.color_format() = dst_format;
                 return res;
             }
+            if ((context.color_format() == ColorFormat::RGB || context.color_format() == ColorFormat::BGR) &&
+                (dst_format == ColorFormat::GRAY)) {
+                auto node = nodes[0];
+                auto elem_type = node.get_element_type();
+                auto shape = node.get_partial_shape();
+                OPENVINO_ASSERT(shape.size() == 4,
+                                "Input shape size should be equal to 4, actual size: ",
+                                shape.size());
+                auto channels_idx = get_and_check_channels_idx(context.layout(), shape);
+                OPENVINO_ASSERT(shape[channels_idx] == 3,
+                                "Channels dimesion should be equal to 3, actual value: ",
+                                shape[channels_idx]);
+
+                auto is_transposed = false, is_converted = false;
+                if (channels_idx + 1 == shape.size()) {
+                    // Transpose N...C  to NC...
+                    auto permutation = layout::utils::find_permutation(context.layout(), shape, ov::Layout{"NC..."});
+                    auto perm_constant =
+                        op::v0::Constant::create<int64_t>(element::i64, Shape{permutation.size()}, permutation);
+                    node = std::make_shared<op::v1::Transpose>(node, perm_constant);
+                    is_transposed = true;
+                }
+                if (elem_type.is_integral_number()) {
+                    // Compute in floats due weights are floats
+                    node = std::make_shared<op::v0::Convert>(node, element::f32);
+                    is_converted = true;
+                }
+
+                auto weights_data = context.color_format() == ColorFormat::RGB
+                                        ? std::vector<float>{0.299f, 0.587f, 0.114f}
+                                        : std::vector<float>{0.114f, 0.587f, 0.299f};
```

Comment:
can we add a reference where we took these numbers?

Reply:
Done, thanks!